### PR TITLE
update logging.cc

### PR DESCRIPTION
**Pull request by HaHaJeff**
原logging.cc中的maybeRotate函数无法完成完成日志名更新

### CODE REVIEW

File: handy/logging.cc [hunk 1]
```diff
@@ -76,8 +76,10 @@ void Logger::setFileName(const string& filename) {
 }
 
 void Logger::maybeRotate() {
+  //以第一次设置的日志文件名为日志基准文件名
+  static std::string basename = filename_;
     time_t now = time(NULL);
-    if (filename_.empty() || (now - timezone) / rotateInterval_ == (lastRotate_ - timezone)/ rotateInterval_) {
+    if (filename_.empty() || (now - timezone) / rotateInterval_ == (lastRotate_ - timezone) / rotateInterval_) {
```
Comment (jovan-wong) [inline handy/logging.cc]: The condition in `if` exp is too confused to read it clearly.
Could you write it in an independent line?

File: handy/logging.cc [hunk 2]
```diff
@@ -89,20 +91,18 @@ void Logger::maybeRotate() {
     struct tm ntm;
     localtime_r(&now, &ntm);
     char newname[4096];
-    snprintf(newname, sizeof(newname), "%s.%d%02d%02d%02d%02d",
-        filename_.c_str(), ntm.tm_year + 1900, ntm.tm_mon + 1, ntm.tm_mday,
-        ntm.tm_hour, ntm.tm_min);
-    const char* oldname = filename_.c_str();
+    snprintf(newname, sizeof(newname), "%s.%d%02d%02d%02d%02d", basename.c_str(), ntm.tm_year + 1900, ntm.tm_mon + 1, ntm.tm_mday, ntm.tm_hour, ntm.tm_min);
```
Comment (jovan-wong) [inline handy/logging.cc]: `newname` -> `new_name`

File: handy/logging.cc [hunk 3]
```diff
@@ -76,8 +76,10 @@ void Logger::setFileName(const string& filename) {
 }
 
 void Logger::maybeRotate() {
+  //以第一次设置的日志文件名为日志基准文件名
+  static std::string basename = filename_;
```
Comment (jovan-wong) [inline handy/logging.cc]: This local static variable may make readers confused.
Add some comments for it?

Comment (HaHaJeff) [reply handy/logging.cc]: I guess that the author's purpose is that rolling the log file after per logInterval_. However, the original function named with mayRotate can't finish it. So, i add the static variable to record the first filename_, meanwhile, make the value as a basename. The real log filename maybe constructed by basename and current time.

File: handy/logging.cc (hunk 1 above)
Comment (HaHaJeff) [reply handy/logging.cc]: the interval of updating log filename is rotateInterval_, the condition in if exp is means that the interval between in two updates whether greater the rotateInterval_.

Comment (jovan-wong) [reply handy/logging.cc]: I meant that for more readability, we should write it like this:
```c++
bool xxxxx =  (now - timezone) / rotateInterval_ == (lastRotate_ - timezone) / rotateInterval_;
if (filename_.empty() ||xxxxx) {
//...
```

File: handy/logging.cc (hunk 3 above)
Comment (jovan-wong) [reply handy/logging.cc]: Also, I meant add some comments in the line before `static .....` to explain why we use  local static variable here.

**Comment by jovan-wong** [comment]
@yedf cc

**Comment by jovan-wong** [comment]
There is a conflicting file. Could u plz resolve it?

**Comment by jovan-wong** [comment]
FYI @yedf  The ci is crashed.

File: handy/logging.cc (hunk 3 above)
Comment (HaHaJeff) [reply handy/logging.cc]: Sorroy, I understand your suggestion wrong.  besides, I will finish it as soon as possible.

Comment (HaHaJeff) [reply handy/logging.cc]: The best way to solve this problem is changing class member filename_ to basename_, and change function setFileName(const std::string&) to setBaseName(const std::string&). However, the simplest way to meet the demand is adding a local static variable. It doesn't change the strcuture of Logger class.

Comment (jovan-wong) [reply handy/logging.cc]: I think changing `filename_` to `basename_` will make more sense. 
We could do everything that make our codes better, right?

Comment (HaHaJeff) [reply handy/logging.cc]: I'm so sorry. After testing, i found the original function can work well.

Comment (jovan-wong) [reply handy/logging.cc]: Fine. I will close this PR.

**Pull request closed without merging**